### PR TITLE
renaming transforms -> transformers

### PR DESCRIPTION
**Pull request by sfilipi**
Addresses part of #1318 renaming transforms to transformers. 
Some addl cleanup.

### CODE REVIEW

**Comment by sfilipi** [comment]
@Ivanidzo4ka, @Zruty0  do we want one estimator/transformer per file, or not necessary? WRT file names, if both the estimator and the transformer are in the same page, I have been keeping the file name just to "ActionPerforming" ex. KeyToValueMapping.  LMK if anything else is preferred.

File: src/Microsoft.ML.Data/Transforms/TermTransformer.cs [hunk 1]
```diff
@@ -42,7 +42,7 @@ namespace Microsoft.ML.Transforms.Categorical
     // * The Key value is the one-based index of the item in the dictionary.
     // * Not found is assigned the value zero.
     /// <include file='doc.xml' path='doc/members/member[@name="TextToKey"]/*' />
-    public sealed partial class TermTransform : OneToOneTransformerBase
+    public sealed partial class TermTransformer : OneToOneTransformerBase
```
Comment (artidoro) [inline src/Microsoft.ML.Data/Transforms/TermTransformer.cs]: >TermTransformer [](start = 32, length = 15)

Should we keep TermTransformer or use ValueToKeyMappingTransformer? To make it ActionPerformingTransformer

File: src/Microsoft.ML.Data/Transforms/LabelIndicatorTransform.cs [hunk 2]
```diff
@@ -12,18 +12,18 @@
...
+    LabelIndicatingTransformer.LoaderSignature)]
+[assembly: LoadableClass(typeof(void), typeof(LabelIndicatingTransformer), null, typeof(SignatureEntryPointModule), LabelIndicatingTransformer.LoadName)]
 
 namespace Microsoft.ML.Transforms
 {
     /// <summary>
     /// Remaps multiclass labels to binary T,F labels, primarily for use with OVA.
     /// </summary>
-    public sealed class LabelIndicatorTransform : OneToOneTransformBase
+    public sealed class LabelIndicatingTransformer : OneToOneTransformBase
```
Comment (artidoro) [inline src/Microsoft.ML.Data/Transforms/LabelIndicatorTransform.cs]: I am not convinced that we should use indicating as a verb. I don't know if indicating gives the right idea. An indicator is a common concept, but turning it into indicating might remove from that concept. Maybe binarizing? Or maybe keeping indicator, and adding mapping?

File: src/Microsoft.ML.Data/Transforms/TrainAndScoreTransformer.cs [hunk 3]
```diff
@@ -11,15 +11,15 @@
...
 
-[assembly: LoadableClass(TrainAndScoreTransform.Summary, typeof(IDataTransform), typeof(TrainAndScoreTransform),
-    typeof(TrainAndScoreTransform.Arguments), typeof(SignatureDataTransform), "Train and Score Predictor", "TrainScore")]
+[assembly: LoadableClass(TrainAndScoreTransformer.Summary, typeof(IDataTransform), typeof(TrainAndScoreTransformer),
+    typeof(TrainAndScoreTransformer.Arguments), typeof(SignatureDataTransform), "Train and Score Predictor", "TrainScore")]
 
 namespace Microsoft.ML.Transforms
 {
-    public static class ScoreTransform
+    public static class ScoreTransformer
```
Comment (artidoro) [inline src/Microsoft.ML.Data/Transforms/TrainAndScoreTransformer.cs]: Maybe ScoringTransformer? To make it ActionPerformingTransformer

File: src/Microsoft.ML.Data/Transforms/SelectColumns.cs [hunk 4]
```diff
@@ -14,30 +14,30 @@
 using System.Collections.Generic;
 using System.Linq;
 
-[assembly: LoadableClass(SelectColumnsTransform.Summary, typeof(IDataTransform), typeof(SelectColumnsTransform),
-                typeof(SelectColumnsTransform.Arguments), typeof(SignatureDataTransform),
-                SelectColumnsTransform.UserName, "SelectColumns", "SelectColumnsTransform", SelectColumnsTransform.ShortName, DocName = "transform/SelectTransforms.md")]
+[assembly: LoadableClass(SelectColumnsTransformer.Summary, typeof(IDataTransform), typeof(SelectColumnsTransformer),
+                typeof(SelectColumnsTransformer.Arguments), typeof(SignatureDataTransform),
```
Comment (shmoradims) [inline src/Microsoft.ML.Data/Transforms/SelectColumns.cs]: >SelectColumns [](start = 23, length = 13)

ColumnsSelecting?

**Comment by shmoradims** [comment]
> // Licensed to the .NET Foundation under one or more agreements.

Similar issue to the other PR. Filenames don't follow a pattern. Sometimes they end in Transformer sometimes they don't.

---
Refers to: src/Microsoft.ML.Data/Transforms/KeyToValue.cs:1 in 69b883e. [](commit_id = 69b883e46fab58d94429a23d9f0a8e1ad4fa7276, deletion_comment = False)

**Comment by sfilipi** [comment]
> // Licensed to the .NET Foundation under one or more agreements.

see my question to Pete and Ivan. There is a pattern: when there is only the estimator or only the transformer in the file, the file takes its full name. If both are in the same file i just left the name to ActionPerforming. 

---
In reply to: [438414934](https://github.com/dotnet/machinelearning/pull/1606#issuecomment-438414934) [](ancestors = 438414934)

---
Refers to: src/Microsoft.ML.Data/Transforms/KeyToValueTransform.cs:1 in 69b883e. [](commit_id = 69b883e46fab58d94429a23d9f0a8e1ad4fa7276, deletion_comment = False)

**Comment by Ivanidzo4ka** [comment]
Do I personally prefer to have one file for one class (with exception of nested classes and really small basic classes) - Yes. 
Do I think it should be part of code standards - maybe
Should you do it  now?  - Don't we have better tasks for your valuable time?

---
In reply to: [438147583](https://github.com/dotnet/machinelearning/pull/1606#issuecomment-438147583) [](ancestors = 438147583)

**Comment by sfilipi** [comment]
> // Licensed to the .NET Foundation under one or more agreements.

Marking this as resolved, to track desired behavior in the other comment. 

---
In reply to: [438481297](https://github.com/dotnet/machinelearning/pull/1606#issuecomment-438481297) [](ancestors = 438481297,438414934)

---
Refers to: src/Microsoft.ML.Data/Transforms/KeyToValueTransform.cs:1 in 69b883e. [](commit_id = 69b883e46fab58d94429a23d9f0a8e1ad4fa7276, deletion_comment = False)

**Comment by sfilipi** [comment]
eh... :)

---
In reply to: [438481328](https://github.com/dotnet/machinelearning/pull/1606#issuecomment-438481328) [](ancestors = 438481328,438147583)

**Comment by Zruty0** [comment]
I think it's better to have one pair of transformer/estimator per file. Going through existing files and fixing that is not necessary right now.

---
In reply to: [438482792](https://github.com/dotnet/machinelearning/pull/1606#issuecomment-438482792) [](ancestors = 438482792,438481328,438147583)

File: src/Microsoft.ML.Data/Transforms/ExtensionsCatalog.cs [hunk 5]
```diff
@@ -85,8 +85,8 @@ public static ColumnSelectingEstimator DropColumns(this TransformsCatalog catalo
         public static ColumnSelectingEstimator SelectColumns(this TransformsCatalog catalog,
```
Comment (Zruty0) [inline src/Microsoft.ML.Data/Transforms/ExtensionsCatalog.cs]: >SelectColumns [](start = 47, length = 13)

can we remove this altogether? I don't see a value for this. Actually, I would rename `KeepColumns` to `SelectColumns`

File: src/Microsoft.ML.Data/Transforms/LabelIndicatorTransform.cs (hunk 2 above)
Comment (Zruty0) [reply src/Microsoft.ML.Data/Transforms/LabelIndicatorTransform.cs]: This is not even a transformer btw

---
In reply to: [233189819](https://github.com/dotnet/machinelearning/pull/1606#discussion_r233189819) [](ancestors = 233189819)

File: src/Microsoft.ML.Data/Transforms/ExtensionsCatalog.cs (hunk 5 above)
Comment (sfilipi) [reply src/Microsoft.ML.Data/Transforms/ExtensionsCatalog.cs]: If they want to drop some, and keep some in one shot? Also I was following the constructors in the class; if de decide to remove Keep, shall it happen there, and as a separate work item?

---
In reply to: [233621944](https://github.com/dotnet/machinelearning/pull/1606#discussion_r233621944) [](ancestors = 233621944)

**Comment by sfilipi** [comment]
@artidoro, @shmoradims would you have time to look at this today?

**Pull request merged**